scipy/linalg/autosol.py

```diff
@@ -0,0 +1,405 @@
+from math import sqrt
+import numpy as np
+from numpy import atleast_1d, atleast_2d
+# import scipy.linalg.decomp_svd
+from scipy.linalg.decomp import _asarray_validated
+# from scipy.linalg.misc import LinAlgError, _datacopied, LinAlgWarning
+from scipy.linalg.misc import LinAlgError
+
+# Author : Rondall E. Jones, August 2020
+
+
+def two_norm(x):
+    """ Computes two-norm of 1-D array x without overflow"""
+    ln = len(x)
+    if ln == 0:
+        return 0.0
+    y = abs(x)
+    mx = max(y)
+    if mx == 0.0:
+        return 0.0
+    sm = 0.0
+    for i in range(0, ln):
+        sm += (x[i] / mx) ** 2
+    return mx * sqrt(sm)
+
+
+def rms(x):
+    """ Computes root-mean-square of 1-D array x without overflow"""
+    s = two_norm(x)
+    return s / sqrt(len(x))
+
+
+def compute_mov_sums(g, w, m):

```

Comment:
Also here you can use convolution as in `np.convolve(g ,np.ones(w, dtype=int),'valid')` or whichever behavior you prefer (see `numpy.convolve` docstring)